frontend/components/gameBox/SmartGameBoxEnvironmentList.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import styled from 'styled-components';
+import { FetchGame } from '../../storage/utils/fetchGameData';
+import { Environment } from '../../storage/models/Environment';
+
+const Item = styled.a`
+	font-size: 12px;
+	margin-left: 20px;
+	display: list-item;
+	list-style: circle;
+`;
+
+interface SmartGameBoxEnvironmentListState {
+	isDataAvailable: boolean;
+	environments: Array<Environment> | null;
+}
+
+interface SmartGameBoxEnvironmentListProps {
+	gameId: number;
+}
+
+export class SmartGameBoxEnvironmentList extends React.Component<SmartGameBoxEnvironmentListProps, SmartGameBoxEnvironmentListState> {
+	constructor(props: SmartGameBoxEnvironmentListProps) {
+		super(props);
+		this.state = {
+			isDataAvailable: false,
+			environments: null,
+		};
+	}
+	public async componentDidMount() {
+		try {
+			this.setState({ environments: await FetchGame.environments(this.props.gameId), isDataAvailable: true });
+		} catch (error) {
+			console.log(error);
+			this.setState({ isDataAvailable: false });
+		}
+	}
+	public render() {
+		if (this.state.isDataAvailable) {
+			return (
+				<>
```

Comment:
Ditto here, extract the actual rendering to a separate component. The idea is for the Smart... components to be a wrapper that does nothing except add logic on top of the component.